extensions/interactions/base_test.py

```diff
@@ -246,6 +247,142 @@ def test_interaction_rule_descriptions_in_dict(self):
             'IsWithinTolerance': 'is within {{tol|Real}} of {{x|Real}}'
         })
 
+    def test_html_field_types_to_rule_specs_mapping_are_valid(self):
+        """Test that the structure of the file html_field_types_to_rule_specs.
+        json are valid. This test ensures that whenever any new type of
+        interaction or rule type with HTML string is added, the file
+        html_field_types_to_rule_specs.json should be updated accordingly.
+        """
+        # The file having the information about the assembly of the html in the
+        # rule specs.
+        html_field_types_to_rule_specs_dict = json.loads(
+            utils.get_file_contents(
+                feconf.HTML_FIELD_TYPES_TO_RULE_SPECS_FILE_PATH))
+
+        # The file having the templates for the structure of the rule specs.
+        # Contents of the file html_field_types_to_rule_specs.json will be
+        # verified against this file.
+        rule_descriptions_dict = json.loads(
+            utils.get_file_contents(feconf.RULES_DESCRIPTIONS_FILE_PATH))
+
+
+        # In the following part, we generate the html_field_types_to_rule_specs
+        # dict based on the values in the rule_descriptions.json file.
+        generated_html_field_types_dict = {}
+        html_types = (
```

Comment:
This should not exist. Again please don't hardcode anything. Use rule_descriptions_dict to programmatically compute **all** the info you need.

Reply:
@seanlip One reason this is done is: that the value for the key `format` cannot be derived from the `rule_descriptions_dict`, and we have to associate each type of format to a html_type.
So even if I calculate the html_types from rule_descriptions_dict, I would need to check the each html_type and assign a format accordingly. Do you have any  suggestions for this?

Reply:
Ah, I see the issue -- thanks for explaining clearly. Yes, the issue is that we're trying to map the variable type to a "how to process it" method.

This is therefore the part that you can't generate fully. Your validation will need to check everything except that "format" key. It should also check that each interior dictionary has a "format" key and that that key is one of a small set of allowed values (which should be named constants).

So perhaps what you can do is: start by iterating through the actual html field types dict and asserting that such a key exists in each inner dict, and that its value satisfies the containment criteria, then remove it (or, perhaps better, replace it by a unique sentinel value). Then generate the rest of the structure from the other json file and verify equality.

Reply:
done